WPMedia/fusion-news-theme-blocks
# TMEDIA-186 - Navigation blocks content filtering

## Description

At content filtering to navigation blocks

* Navigation Chain
* Links Bar Block
* Footer block

## Jira Ticket
- [TMEDIA-186](https://arcpublishing.atlassian.net/browse/TMEDIA-186)

## Acceptance Criteria
1. Feature-level content filtering is applied to Navigation blocks, so that only the data that is actually used in displaying the block is included: 
    * Navigation Chain
    * Links Bar Block
    * Footer Block
2. Tests and documentation are written to cover this functionality

## Test Steps

1. Checkout this branch `git checkout TMEDIA-186-navigation-blocks-content-filtering`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/footer-block,@wpmedia/header-nav-chain-block,@wpmedia/links-bar-block`
3. On a new page add the three blocks (Navigation Chain, Links Bar, Footer) and configure their content config
    * Navigation Chain
        * Sections Menu hierarchy = hamburger-menu
        * Horizontal Links hierarchy = links-bar
    * Links Bar
        * hierarchy = links-bar
    * Footer
        * hierarchy = footer
4. Publish page and verify the blocks still work as intended
5. Verify content cache is small using Fusion bookmarklet

## Effect Of Changes
### Before

<img width="1279" alt="TMEDIA-186-before" src="https://user-images.githubusercontent.com/868127/113284943-443c5180-92e2-11eb-9b71-f2b7e6e1ccef.png">


### After

<img width="1279" alt="TMEDIA-186-after" src="https://user-images.githubusercontent.com/868127/113284965-4acac900-92e2-11eb-936b-c2a102bfe3c2.png">


## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
